charts/Makefile

```diff
@@ -19,8 +19,16 @@ build: clean
 	cp -r gadget $(BUILD_DIR)
 	cp -r ../pkg/resources/crd/bases $(CHART_DIR)/crds
 	mv $(CHART_DIR)/Chart.yaml.tmpl $(CHART_DIR)/Chart.yaml
+	# sed on macOS requires an extra argument for the backup file
+	# or an empty string to disable backup files, GNU sed does not
+	# require this argument but allows it
+ifeq ($(shell uname -s),Darwin)
```

Comment:
I wonder if testing `uname -s` will cause problems because it assumes the version of sed based on the kernel. For example, on Freebsd, or if GNU sed has been installed on Darwin, etc.

Inspired from [stackoverflow](https://stackoverflow.com/questions/2320564/sed-i-command-for-in-place-editing-to-work-with-both-gnu-sed-and-bsd-osx):
```Makefile
# the --version option is only found in the GNU version of sed:
ifeq ($(shell sed --version >/dev/null 2>&1 ; echo $$?),0)
        @echo Using GNU sed
        sed -i "s/..." file
else
        @echo Using BSD sed
        sed -i "" "s/..." file
endif
```

Reply:
after looking around my I suggest 

```makefile
SED_VERSION := $(shell sed --version 2>&1)

ifeq ($(findstring GNU,$(SED_VERSION)),GNU)
	@echo "Using GNU sed"
	sed -i "s/%VERSION%/$(CHART_VERSION)/g" $(CHART_DIR)/Chart.yaml
	sed -i "s/%APP_VERSION%/$(APP_VERSION)/g" $(CHART_DIR)/Chart.yaml
else
	@echo "Using BSD sed"
	sed -i "" "s/%VERSION%/$(CHART_VERSION)/g" $(CHART_DIR)/Chart.yaml
	sed -i "" "s/%APP_VERSION%/$(APP_VERSION)/g" $(CHART_DIR)/Chart.yaml
endif
```